ui/src/components/Filter.tsx

```diff
@@ -79,7 +72,7 @@ export function Filter({state, setState}): JSX.Element {
           id="filter-autocomplete"
           options={countries || []}
           style={{ width: 312 }}
-          onChange={onTagsChange}
+          onChange={(event, value) => onChange(value)}
```

Comment:
Let's change the type here. This means that:

```suggestion
          onChange={onChange}
```

```ts
export interface FilterProps {
  onChange: (event: idkWhatEventTypeIs, value: string[]) => void
}
```

Then you'll have to change:

```ts
// ui/src/pages/ShoeMap.tsx 
function onTagsChange(_, value: string[]) {
  setFilteredCountries(options);
}
```